pvlib/tests/test_irradiance.py

```diff
@@ -1129,10 +1130,30 @@ def test_dni():
     ghi = pd.Series([90, 100, 100, 100, 100])
     dhi = pd.Series([100, 90, 50, 50, 50])
     zenith = pd.Series([80, 100, 85, 70, 85])
-    clearsky_dni = pd.Series([50, 50, 200, 50, 300])
+    dni_clear = pd.Series([50, 50, 200, 50, 300])
 
     dni = irradiance.dni(ghi, dhi, zenith,
-                         clearsky_dni=clearsky_dni, clearsky_tolerance=2)
+                         dni_clear=dni_clear, clearsky_tolerance=2)
+    assert_series_equal(dni,
+                        pd.Series([float('nan'), float('nan'), 400,
+                                   146.190220008, 573.685662283]))
+
+    dni = irradiance.dni(ghi, dhi, zenith)
+    assert_series_equal(dni,
+                        pd.Series([float('nan'), float('nan'), 573.685662283,
+                                   146.190220008, 573.685662283]))
+
+
+@fail_on_pvlib_version("0.12")
+def test_dni_dni_clearsky_deprecation():
+    ghi = pd.Series([90, 100, 100, 100, 100])
+    dhi = pd.Series([100, 90, 50, 50, 50])
+    zenith = pd.Series([80, 100, 85, 70, 85])
+    dni_clear = pd.Series([50, 50, 200, 50, 300])
+
+    with pytest.warns(pvlibDeprecationWarning, match='dni_clear'):
+        dni = irradiance.dni(ghi, dhi, zenith,
+                             clearsky_dni=dni_clear, clearsky_tolerance=2)
     assert_series_equal(dni,
                         pd.Series([float('nan'), float('nan'), 400,
                                    146.190220008, 573.685662283]))
```

Comment:
No need to check the return values for these deprecation tests IMHO.  The point is to verify that the warning is raised.